rotkehlchen/chain/ethereum/modules/aave/v2/decoder.py

```diff
@@ -0,0 +1,253 @@
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.notes = notes
+                else:
+                    event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    resolved_asset = event.asset.resolve_to_asset_with_symbol()
+                    event.notes = f'Return {amount} {resolved_asset.symbol} to AAVE v2'  # noqa: E501
+                # Set counterparty for both events
+                event.counterparty = CPT_AAVE_V2
+
+    def _decode_borrow(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        """Decode aave v2 borrow event"""
+        on_behalf_of = hex_or_bytes_to_address(tx_log.topics[2])
+        user = hex_or_bytes_to_address(tx_log.data[:32])
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data[32:64]),
+            asset=token,
+        )
+        rate_mode = hex_or_bytes_to_int(tx_log.data[64:96])
+        rate = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(tx_log.data[96:128]),
+            token_decimals=25,
+        )
+        notes = f'Borrow {amount} {token.symbol} from AAVE v2 user {user} with {"stable" if rate_mode == 1 else "variable"} APY {rate}'  # noqa: E501
+        if on_behalf_of != user:
+            notes += f' on behalf of {on_behalf_of}'
+        for event in decoded_events:
+            if (
+                event.balance.amount == amount and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == user and
+                event.event_type == HistoryEventType.RECEIVE
+            ):
+                if event.asset == token:
+                    event.event_subtype = HistoryEventSubType.GENERATE_DEBT
+                    event.notes = notes
+                else:
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    resolved_asset = event.asset.resolve_to_asset_with_symbol()
+                    event.notes = f'Receive {amount} {resolved_asset.symbol} from AAVE v2'  # noqa: E501
+                # Set counterparty for both events
+                event.counterparty = CPT_AAVE_V2
+
+    def _decode_repay(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        """Decode aave v2 repay event"""
+        user = hex_or_bytes_to_address(tx_log.topics[2])
+        repayer = hex_or_bytes_to_address(tx_log.topics[3])
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data),
+            asset=token,
+        )
+        notes = f'Repay {amount} {token.symbol} on AAVE v2'
+        if repayer != user:
+            notes += f' from {repayer} for {user}'
```

Comment:
same comment as above. Check who is in the location label and add the address reference only if it's not